File: GeneratedEntityReaderClasses/Components/PhysicsImageShapeComponent.cs
namespace GeneratedNoitaClasses;
public class PhysicsImageShapeComponent : NoitaComponentBase,  iNoitaType<PhysicsImageShapeComponent>{
	public NoitaBool Deleted { get; set; } = new();
	public NoitaBool Enabled { get; set; } = new();
	public NoitaString Tags { get; set; } = new();
	public NoitaBool is_root { get; set; } = new();
	public NoitaInt body_id { get; set; } = new();
	public NoitaBool use_sprite { get; set; } = new();
	public NoitaBool is_circle { get; set; } = new();
	public NoitaBool centered { get; set; } = new();
	public NoitaFloat offset_x { get; set; } = new();
	public NoitaFloat offset_y { get; set; } = new();
	public NoitaFloat z { get; set; } = new();
	public NoitaString image_file { get; set; } = new();
	public NoitaInt material { get; set; } = new();

    public void Read(NoitaStream s){
		Deleted.Read(s);
		Enabled.Read(s);
		Tags.Read(s);
		is_root.Read(s);
		body_id.Read(s);
		use_sprite.Read(s);
		is_circle.Read(s);
		centered.Read(s);
		offset_x.Read(s);
		offset_y.Read(s);
		z.Read(s);
		image_file.Read(s);
		material.Read(s);

    }
    public void Write(NoitaStream s){
		Deleted.Write(s);
		Enabled.Write(s);
		Tags.Write(s);
		is_root.Write(s);
		body_id.Write(s);
		use_sprite.Write(s);
		is_circle.Write(s);
		centered.Write(s);
		offset_x.Write(s);
		offset_y.Write(s);
		z.Write(s);
		image_file.Write(s);
		material.Write(s);

    }
}

File: GeneratedEntityReaderClassesPython/GeneratedClasses/Components/PhysicsImageShapeComponent.py
from dataclasses import dataclass, field
from BasicNoitaTypes import NoitaBool
from BasicNoitaTypes import NoitaString
from BasicNoitaTypes import NoitaInt
from BasicNoitaTypes import NoitaFloat

from NoitaStream import NoitaStream
@dataclass
class PhysicsImageShapeComponent:
    Deleted:NoitaBool = field(default_factory=NoitaBool)
    Enabled:NoitaBool = field(default_factory=NoitaBool)
    Tags:NoitaString = field(default_factory=NoitaString)
    is_root:NoitaBool = field(default_factory=NoitaBool)
    body_id:NoitaInt = field(default_factory=NoitaInt)
    use_sprite:NoitaBool = field(default_factory=NoitaBool)
    is_circle:NoitaBool = field(default_factory=NoitaBool)
    centered:NoitaBool = field(default_factory=NoitaBool)
    offset_x:NoitaFloat = field(default_factory=NoitaFloat)
    offset_y:NoitaFloat = field(default_factory=NoitaFloat)
    z:NoitaFloat = field(default_factory=NoitaFloat)
    image_file:NoitaString = field(default_factory=NoitaString)
    material:NoitaInt = field(default_factory=NoitaInt)
    
    def Read(self, s:NoitaStream):
        self.Deleted.Read(s)
        self.Enabled.Read(s)
        self.Tags.Read(s)
        self.is_root.Read(s)
        self.body_id.Read(s)
        self.use_sprite.Read(s)
        self.is_circle.Read(s)
        self.centered.Read(s)
        self.offset_x.Read(s)
        self.offset_y.Read(s)
        self.z.Read(s)
        self.image_file.Read(s)
        self.material.Read(s)
    
    def Write(self, s:NoitaStream):
        self.Deleted.Write(s)
        self.Enabled.Write(s)
        self.Tags.Write(s)
        self.is_root.Write(s)
        self.body_id.Write(s)
        self.use_sprite.Write(s)
        self.is_circle.Write(s)
        self.centered.Write(s)
        self.offset_x.Write(s)
        self.offset_y.Write(s)
        self.z.Write(s)
        self.image_file.Write(s)
        self.material.Write(s)

